extensions/amp-story/1.0/animation.js

```diff
@@ -173,34 +195,50 @@ class AnimationRunner {
   }
 
   /**
-   * @param {!KeyframesOrFilterFnDef} keyframesArrayOrFn
-   * @return {!Promise<!KeyframesDef>}
+   * Evaluates a preset's keyframes function using dimensions.
+   * @param {!WebKeyframesDef|!WebKeyframesCreateFnDef} keyframesOrCreateFn
+   * @return {!Promise<!WebKeyframesDef>}
    * @private
    */
-  filterKeyframes_(keyframesArrayOrFn) {
-    if (Array.isArray(keyframesArrayOrFn)) {
-      return Promise.resolve(keyframesArrayOrFn);
+  resolvePresetKeyframes_(keyframesOrCreateFn) {
+    if (typeof keyframesOrCreateFn !== 'function') {
+      return Promise.resolve(keyframesOrCreateFn);
     }
-    return this.getDims().then(keyframesArrayOrFn);
+    return this.getDims().then(keyframesOrCreateFn);
   }
 
   /**
-   * @return {!Promise<
-   *   !../../amp-animation/0.1/web-animation-types.WebKeyframeAnimationDef>}
-   * @private
+   * Normalizes an animation definition into a WebAnimationDef to consume.
+   * @param {!StoryAnimationDef|!WebAnimationDef} animationDef
+   * @return {!Promise<!WebAnimationDef>}
    */
-  getWebAnimationDef_() {
-    return this.keyframes_.then((keyframes) => ({
+  getWebAnimationDef_(animationDef) {
+    const {preset} = animationDef;
+    if (!preset) {
+      // This is an amp-animation config, so it's already formed how the
+      // WebAnimations Builder wants it.
+      return Promise.resolve(/** @type {!WebAnimationDef} */ (animationDef));
+    }
+    const {target, delay, duration, easing} = animationDef;
+    return this.resolvePresetKeyframes_(preset.keyframes).then((keyframes) => ({
+      target,
+      easing,
       keyframes,
-      target: this.target_,
-      delay: `${this.delay_}ms`,
-      duration: `${this.duration_}ms`,
-      easing: this.easing_,
+      delay: `${delay}ms`,
```

Comment:
These should be accepting values w/o `ms`, I believe. Not sure if essentialy, but I believe there's a way to trigger the time rescaling logic that could be useful.

Reply:
Nice!